### PR TITLE
Add separate service account to each agent

### PR DESCRIPTION
**Pull request by makoscafee**

### CODE REVIEW

File: stable/insights-agent/templates/kubesec/serviceaccount.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{{- if .Values.kubesec.enabled -}}
```
Comment (baderbuddy) [inline stable/insights-agent/templates/kubesec/serviceaccount.yaml]: I'd say let's not bother with an individual service account when there's no RBAC, but I'd be happy to hear other thoughts on this.

File: stable/insights-agent/templates/kube-bench/serviceaccount.yaml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+{{- if .Values.kubebench.enabled -}}
```
Comment (baderbuddy) [inline stable/insights-agent/templates/kube-bench/serviceaccount.yaml]: Whatever is decided with kubesec below would apply here too.

**Review by baderbuddy — COMMENTED** [review]
Looks pretty good, have you tested it out? Also I have a question about whether or not it's okay to have a generic service account when there's no specific permissions granted.

File: stable/insights-agent/templates/kubesec/serviceaccount.yaml (hunk 1 above)
Comment (rbren) [reply stable/insights-agent/templates/kubesec/serviceaccount.yaml]: I think we definitely need some RBAC for kubesec, it `kubectl get`s a bunch of controllers.

File: stable/insights-agent/templates/kube-bench/serviceaccount.yaml (hunk 2 above)
Comment (rbren) [reply stable/insights-agent/templates/kube-bench/serviceaccount.yaml]: I agree - we can probably drop the SA here (as well as `serviceAccountName` in the podspec)

File: stable/insights-agent/templates/kubesec/serviceaccount.yaml (hunk 1 above)
Comment (makoscafee) [reply stable/insights-agent/templates/kubesec/serviceaccount.yaml]: `kubesec` uses/gets/list these resources `deployments,statefulsets,daemonsets,namespaces`

**Comment by rbren** [comment]
@baderbuddy IIUC if we don't specify a particular service account, it'll use the "default service account", which is set cluster-to-cluster. I think we should probably specify in that case

File: stable/insights-agent/templates/kube-hunter/rbac.yaml [hunk 3]
```diff
@@ -0,0 +1,49 @@
+{{- if .Values.kubehunter.enabled -}}
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: {{ include "insights-agent.fullname" . }}-kube-hunter
+  labels:
+    app: insights-agent
+---
+apiVersion: rbac.authorization.k8s.io/v1beta1
+kind: ClusterRole
```
Comment (rbren) [inline stable/insights-agent/templates/kube-hunter/rbac.yaml]: I think we can delete this ClusterRole and ClusterRoleBinding - kube-hunter doesn't seem to need any RBAC. Let's just use an empty service account

File: stable/insights-agent/templates/kube-bench/cronjob.yaml [hunk 4]
```diff
@@ -2,7 +2,6 @@
 {{- $_ := set . "Label" "kube-bench" -}}
 {{- $_ := set . "Config" .Values.kubebench -}}
 restartPolicy: Never
-serviceAccountName: {{ include "insights-agent.fullname" . }}
```
Comment (rbren) [inline stable/insights-agent/templates/kube-bench/cronjob.yaml:5]: Let's create and use an empty service account here, like in kube-hunter

**Review by rbren — APPROVED** [review]

**Pull request merged**